pilopt/src/lib.rs

```diff
@@ -427,6 +435,81 @@ fn simplify_expression_single<T: FieldElement>(e: &mut AlgebraicExpression<T>) {
     }
 }
 
+fn try_simplify_associative_operation<T: FieldElement>(
+    left: &AlgebraicExpression<T>,
+    right: &AlgebraicExpression<T>,
+    op: AlgebraicBinaryOperator,
+) -> Option<AlgebraicExpression<T>> {
```

Comment:
Is it possibly to avoid the cloning? The expressions might be quite large, and we might do it recursively for all sub-expressions.